modules/sdcore-control-plane-k8s/README.md

```diff
@@ -0,0 +1,103 @@
+# SD-Core Control Plane Terraform Module
+
+This folder contains the [Terraform][Terraform] module for the [sdcore-control-plane-k8s][sdcore-control-plane-k8s].
+
+The module uses the [Terraform Juju provider][Terraform Juju provider] to model the charm deployment onto any Kubernetes environment managed by [Juju][Juju].
+
+The module can be used to deploy the `sdcore-control-plane-k8s` separately as well as a part of `sdcore-k8s` module, depending on the deployment architecture.
+
+## Module structure
+
+- **main.tf** - Defines the Juju application to be deployed.
+- **variables.tf** - Allows customization of the deployment. Except for exposing the deployment options (Juju model name, channel) also allows overwriting charm's default configuration.
+- **terraform.tf** - Defines the Terraform provider.
+
+## Deploying sdcore-control-plane-k8s module separately
+
+### Pre-requisites
+
+The following tools needs to be installed and should be running in the environment. Please [set up your environment][set-up-environment] before deployment.
+
+- A Kubernetes cluster with the `Multus` and `Metallb` addon enabled.
+- Juju 3.x
+- Juju controller bootstrapped onto the K8s cluster
+- Terraform
+
+### Deploying sdcore-control-plane-k8s with Terraform
+
+Form inside the `sdcore-control-plane-k8s` module folder, initialize the provider:
+
+```console
+terraform init
+```
+
+While creating the plan, the default configuration can be overwritten with `-var-file`. To do that, create a `terraform.tfvars` including the contents similar to below:
+
+```yaml
+# Mandatory Config Options
+model_name = "put your model-name here"
+
+# Optional Configurations
+
+# channel                        = "put the channel for the SD-Core charms here"
+# mongo_channel                  = "put the MongoDB charm channel here"
+# grafana_channel                = "put the Grafana charm channel here"
```

Comment:
It's Grafana agent actually. It's like a worker that collects the data for Grafana.

Reply:
It's being replaced as `grafana-agent`